core/src/main/kotlin/org/evomaster/core/problem/rest/resource/ResourceToken.kt

```diff
@@ -0,0 +1,93 @@
+package org.evomaster.core.problem.rest.resource
+
+/**
+ * this is used for text and name analysis with nl parser
+ * @property originalText is original text before processing the analysis
+ * @property lemma of [originalText]
+ * @property assuredVerb presents whether the token is a verb
+ */
+open class RToken(
+    val originalText : String,
+    protected val lemma : String,
+    var assuredVerb : Boolean = false
+){
+    var isCollection: Boolean =  !assuredVerb && lemma != originalText
+        private set
+
+    fun equals(other : RToken):Boolean{
+        return isCollection == other.isCollection && lemma == other.lemma
+    }
+
+    fun equals(text: String) : Boolean{
+        return text.toLowerCase() == lemma.toLowerCase() || text.toLowerCase() == originalText.toLowerCase()
```

Comment:
i think this could be optimized / simplified.
if i remember correctly, the .equals in Kotlin as a extra boolean parameter to specify if case should be ignored

Reply:
fixed. the function is redundant and removed now.